libs/dal/src/repositories/base-repository.ts

```diff
@@ -87,14 +87,20 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement = object> {
   }
 
   private calcExpireDate(modelName: string, data: FilterQuery<T_DBModel> & T_Enforcement) {
-    const now: number = Date.now();
+    let startDate: Date = new Date();
+    if (data.expireAt) {
+      startDate = new Date(data.expireAt);
+    }
+
     switch (modelName) {
       case 'Message':
         if (data.channel === 'in_app') {
-          return addMonths(now, 6);
+          return addMonths(startDate, 6);
         } else {
-          return addMonths(now, 1);
+          return addMonths(startDate, 1);
         }
+      case 'Notification':
+        return addMonths(startDate, 1);
```

Comment:
Thats because, for jobs and execution details, I am taking the expireAt date of the notification. Which already has the month added. I want the notification, and all records that are related to it to have same expiration date.

Reply:
ahhh I see now, it's hard to notice that because we have that logic hidden in the repo

Reply:
Yes🙈 Do you have a suggestion for me how I can make it clearer ?

Reply:
not sure, maybe easier to read would be to have it in the `create-notification-jobs.usecase.ts` `calculateExpireAt` function instead of repo, because that the place where we calculate the base date, so I don't expect that one to be overriden?